src/agentscope/agents/agent.py

```diff
@@ -2,20 +2,29 @@
 """ Base class for Agent """
 
 from __future__ import annotations
+from abc import ABCMeta
 from typing import Optional
 from typing import Sequence
 from typing import Union
 from typing import Any
 from typing import Callable
-
 from loguru import logger
 
-from .operator import Operator
-from ..models import load_model_by_name
-from ..memory import TemporaryMemory
+from agentscope.agents.operator import Operator
+from agentscope.models import load_model_by_name
+from agentscope.memory import TemporaryMemory
+
+
+class _RecordInitSettingMeta(ABCMeta):
+    """A wrapper to record the init args into `init_settings` field."""
 
+    def __call__(cls, *args: tuple, **kwargs: dict) -> Any:
+        instance = super().__call__(*args, **kwargs)
+        instance.init_settings = {"args": args, "kwargs": kwargs}
```

Comment:
It's suggested to maintain `init_settings` as a private attribute `_init_settings`?